editor/store/persist.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { subscribe, dispatch, getState } from '@wordpress/data';
+
+/**
+ * Adds the rehydratation behavior to redux reducers
+ *
+ * @param {Function}   reducer     The reducer to enhance
+ * @param {String}     reducerKey  The reducer key to persist
+ *
+ * @return {Function}              Enhanced reducer
+ */
+export function withRehydratation( reducer, reducerKey ) {
+	// EnhancedReducer with auto-rehydration
+	const enhancedReducer = ( state, action ) => {
+		const nextState = reducer( state, action );
+
+		if ( action.type === 'REDUX_REHYDRATE' ) {
+			return {
+				...nextState,
+				[ reducerKey ]: action.payload,
+			};
+		}
+
+		return nextState;
+	};
+
+	return enhancedReducer;
+}
+
+/**
+ * Load the initial state and persist on changes
```

Comment:
I think for a separation between title and body there should be a newline between those lines. In a different pull request, @aduth and I discussed not using `@summary`. This should also start with `Loads`.